specs/data_structures.md

```diff
@@ -78,37 +87,36 @@ TODO define a transaction format
 | `blockID`    | [BlockID](#blockid)         |             |
 | `signatures` | [CommitSig](#commitsig)`[]` |             |
 
-## Version
-
-| name                | type     | description |
-| ------------------- | -------- | ----------- |
-| `blockchainVersion` | `uint64` |             |
-| `appVersion`        | `uint64` |             |
-
 ## Time
 
-https://developers.google.com/protocol-buffers/docs/reference/csharp/class/google/protobuf/well-known-types/timestamp
+LazyLedger uses the [Google Protobuf Timestamp](https://developers.google.com/protocol-buffers/docs/reference/csharp/class/google/protobuf/well-known-types/timestamp) format for timestamps, which represents time as seconds in UTC Epoch Time and nanoseconds.
 
 ## BlockID
 
-| name          | type                            | description |
-| ------------- | ------------------------------- | ----------- |
-| `hash`        | [HashDigest](#hashdigest)       |             |
-| `partsHeader` | [PartSetHeader](#partsetheader) |             |
+The block ID is comprised of two distinct Merkle roots:
+1. The root of the [block header](#header)'s fields, in the order provided in the spec.
+1. The root of the complete [serialized](#serialization) block [split into parts](#todo). This is used at the network layer for securely gossiping parts of blocks.
+
+| name          | type                            | description               |
+| ------------- | ------------------------------- | ------------------------- |
+| `headerRoot`  | [HashDigest](#hashdigest)       | Root of the block header. |
+| `partsHeader` | [PartSetHeader](#partsetheader) | Parts header.             |
 
 ## HashDigest
 
 ## Address
 
-## Transaction
+## CodingRoots
+
+## Message
```

Comment:
I'm assuming these are transactions for the LL main sidechain? If so, are these transactions posted directly onto the block, or are they aggregated into a sidechain block like an optimistic rollup?

Reply:
The plan here was for transactions (messages) to be posted individually. Given their consensus-critical nature it shouldn't make any different if they're aggregated or not.

Reply:
They need to be aggregated into state roots so that fraud proofs can be generated however

Reply:
Yes, that's what the `appHash` field in the block header is going to be for.

Reply:
What about intermediate state roots?

Reply:
Couldn't you technically commit to the intermediate state roots and the final state root in a single commitment?

Reply:
No because the intermediate state root need data availability in order to prove fraud, so they need to be included in the data root. Hence why some kind of block aggregation is needed.

Reply:
Ah, good point, I see now.